### DISCUSSION TITLE
Integrating with Pydantic v2

### DISCUSSION THREAD

**Discussion by Hellzed** [sqlalchemy/sqlalchemy#10389] [Usage Questions] [answered]
Documentation mentions that [Pydantic is "not fully compatible" with SQL Alchemy](https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#integrating-with-alternate-dataclass-providers-such-as-pydantic), and according to the git commit history the incompatibility affects both Pydantic v1 and v2.

Are these known issues tracked somewhere, or is there a detailed list?

**Comment by CaselIT** [comment, marked as answer]
Hi,

There are no open issues in sqlalchemy regarding this, since I'm not sure if sqlalchemy can do something to accommodate pydantic. I don't know if there are issues on the Pydantic side.
As far as I know the main issue is on pydantic side, and is that it removes additional attributes on instances that are used by sqlalchemy to track the instances.

If I remember correctly at some point during pydantic 2 beta / rc it seemed that, at least when using dataclasses, was not doing that, but IIRC it appears that the final versions of v2 still behave in the same way.

Not sure if this replies to the question

**Reply by Hellzed** [reply]
Hi, I understand that this is definitely more of a Pydantic issue, I just hoped someone here would have firsthand experience with current Pydantic v2 release behavior. I guess I'll just have to try it and thoroughly test this behavior.

**Reply by CaselIT** [reply]
feel free to post an update if you find something that works
